mlflow/recipes/steps/train.py

```diff
@@ -329,40 +337,69 @@ def my_warn(*args, **kwargs):
                 mlflow.sklearn.log_model(
                     transformer, "transform/transformer", code_paths=self.code_paths
                 )
-                model = make_pipeline(transformer, estimator)
-                if self.predict_proba:
-                    model_schema = infer_signature(
-                        raw_X_train, model.predict_proba(raw_X_train.copy())
-                    )
-                    model_info = mlflow.sklearn.log_model(
-                        model,
-                        f"{self.name}/model",
-                        signature=model_schema,
-                        code_paths=self.code_paths,
-                        pyfunc_predict_fn="predict_proba",
-                    )
-                else:
-                    model_schema = infer_signature(raw_X_train, model.predict(raw_X_train.copy()))
-                    model_info = mlflow.sklearn.log_model(
-                        model,
-                        f"{self.name}/model",
-                        signature=model_schema,
-                        code_paths=self.code_paths,
-                    )
-                output_model_path = get_step_output_path(
+
+                trained_pipeline = make_pipeline(transformer, estimator)
+                # Creating a wrapped recipe model for the pipeline so users can achieve both
+                # predict and predict_proba(for a classification problem) correctly.
+                wrapped_model = WrappedRecipeModel(
+                    self.predict_scores_for_all_classes, self.predict_prefix
+                )
+
+                model_uri = get_step_output_path(
                     recipe_root_path=self.recipe_root,
                     step_name=self.name,
                     relative_path=TrainStep.MODEL_ARTIFACT_RELATIVE_PATH,
                 )
-                if os.path.exists(output_model_path) and os.path.isdir(output_model_path):
-                    shutil.rmtree(output_model_path)
-
-                if self.predict_proba:
-                    mlflow.sklearn.save_model(
-                        model, output_model_path, pyfunc_predict_fn="predict_proba"
+                sklearn_model_uri = get_step_output_path(
+                    recipe_root_path=self.recipe_root,
+                    step_name=self.name,
+                    relative_path=TrainStep.SKLEARN_MODEL_ARTIFACT_RELATIVE_PATH,
+                )
+                if os.path.exists(model_uri) and os.path.isdir(model_uri):
+                    shutil.rmtree(model_uri)
+                if os.path.exists(sklearn_model_uri) and os.path.isdir(sklearn_model_uri):
+                    shutil.rmtree(sklearn_model_uri)
+
+                # Saving the sklearn model along with its pyfunc model since mlflow evaluate
+                # predicts results using an sklearn model
```

Comment:
```suggestion
                # Saving the sklearn model as a separate output since `mlflow.evaluate()`, which is
                # used in the evaluate step of the recipe, needs this model's pyfunc representation to
                # computes metrics (the pyfunc representation of the user-facing model logged to
                # MLflow Tracking is not currently compatible with `mlflow.evaluate()`)
```